app/models/user_seen_subject.rb

```diff
@@ -23,6 +22,15 @@ def self.count_user_activity(user_id, workflow_ids=[])
     scope.sum("cardinality(subject_ids)")
   end
 
+  def self.activity_by_workflow(user_id)
+    scope = self.where(user_id: user_id)
```

Comment:
we should be able to push this to the DB with something like `scope.group(:workflow_id).sum("cardinality(subject_ids)").as_json`